plugins/mongodb/mongodbembed.cpp

```diff
@@ -0,0 +1,2260 @@
+     * @param field RtlFielInfo holds a lot of information about the embed context and here we grab
+     * name of the parameter.
+     * @param query Object holding the bsoncxx::builder::basic::document.
+     */
+    void bindStringParam(unsigned len, const char *value, const RtlFieldInfo * field, std::shared_ptr<MongoDBQuery> query)
+    {
+        query->build()->append(kvp(std::string(field->name), std::string(value, len)));
+    }
+
+    /**
+     * @brief Appends the key value pair to the document. This is used for inserting values into the
+     * query string that gets parsed into a bson document.
+     * 
+     * @param value pointer to value of parameter.
+     * @param field RtlFielInfo holds a lot of information about the embed context. 
+     * @param query Object holding the bsoncxx::builder::basic::document.
+     */
+    void bindBoolParam(bool value, const RtlFieldInfo * field, std::shared_ptr<MongoDBQuery> query)
+    {
+        query->build()->append(kvp(std::string(field->name), bsoncxx::types::b_bool{value}));
+    }
+
+    /**
+     * @brief Appends the key value pair to the document. This is used for inserting values into the
+     * query string that gets parsed into a bson document.
+     * 
+     * @param len Number of chars in value.
+     * @param value pointer to value of parameter.
+     * @param field RtlFielInfo holds a lot of information about the embed context. 
+     * @param query Object holding the bsoncxx::builder::basic::document.
+     */
+    void bindDataParam(unsigned len, const void *value, const RtlFieldInfo * field, std::shared_ptr<MongoDBQuery> query)
+    {
+        size32_t bytes;
+        rtlDataAttr data;
+        rtlStrToDataX(bytes, data.refdata(), len, value);
+
+        query->build()->append(kvp(std::string(field->name), std::string(data.getstr())));
+    }
+
+    /**
+     * @brief Appends the key value pair to the document. This is used for inserting values into the
+     * query string that gets parsed into a bson document.
+     * 
+     * @param value pointer to value of parameter.
+     * @param field RtlFielInfo holds a lot of information about the embed context. 
+     * @param query Object holding the bsoncxx::builder::basic::document.
+     */
+    void bindIntParam(__int64 value, const RtlFieldInfo * field, std::shared_ptr<MongoDBQuery> query)
+    {
+        query->build()->append(kvp(std::string(field->name), bsoncxx::types::b_int64{value}));
+    }
+
+    /**
+     * @brief Appends the key value pair to the document. This is used for inserting values into the
+     * query string that gets parsed into a bson document.
+     * 
+     * @param value pointer to value of parameter.
+     * @param field RtlFielInfo holds a lot of information about the embed context.
+     * @param query Object holding the bsoncxx::builder::basic::document.
+     */
+    void bindUIntParam(unsigned __int64 value, const RtlFieldInfo * field, std::shared_ptr<MongoDBQuery> query)
```

Comment:
Inserting thought here:  Does your README talking about the int vs uint issue?

Reply:
No it does not. There should be a section about type conversion between ECL and MongoDB because there are a few non trivial cases.